### PR TITLE
[IOTDB-2692] Fix compaction exception caused by deleted timeseries

### PR DESCRIPTION
**Pull request by ericpai**
See JIRA: https://issues.apache.org/jira/browse/IOTDB-2692

### CODE REVIEW

**Comment by ericpai** [comment]
@THUMarkLau @choubenson PTAL

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java [hunk 1]
```diff
@@ -125,8 +125,12 @@ private static void compactAlignedSeries(
     List<IMeasurementSchema> measurementSchemas = new ArrayList<>();
     for (String measurement : allMeasurments) {
       // TODO: use IDTable
-      measurementSchemas.add(
-          IoTDB.metaManager.getSeriesSchema(new PartialPath(device, measurement)));
+      PartialPath p = new PartialPath(device, measurement);
+      if (!IoTDB.metaManager.isPathExist(p)) {
+        logger.info("As path {} is dropped we skip it", p.getFullPath());
+        continue;
+      }
```
Comment (qiaojialin) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java:132]: Hi, what if the deletion just takes effect between the check and the getSeriesSchema?

Comment (ericpai) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java:132]: That's a good catch, maybe we can just catch the exception without pre existence check?

**Review by THUMarkLau — COMMENTED** [review]
Please try to add an UT

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java (hunk 1 above)
Comment (qiaojialin) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java:132]: Yes, I prefer to just catch the exception and ignore it

Comment (ericpai) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java:135]: `try-catch` is added instead of pre-check.

**Comment by ericpai** [comment]
> Please try to add an UT

Thanks for your comments! I have updated the existence test cases to cover the situation that timeseries are deleted before compaction.

**Review by qiaojialin — COMMENTED** [review]
@THUMarkLau  please check the MeasurementIterator

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/InnerSpaceCompactionUtils.java [hunk 2]
```diff
@@ -108,12 +112,22 @@ private static void compactNotAlignedSeries(
     while (seriesIterator.hasNextSeries()) {
       checkThreadInterrupted(targetResource);
       // TODO: we can provide a configuration item to enable concurrent between each series
-      String currentSeries = seriesIterator.nextSeries();
+      PartialPath p = new PartialPath(device, seriesIterator.nextSeries());
+      IMeasurementSchema ms;
```
[the remark this answers is not in the document]
Comment (ericpai) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/InnerSpaceCompactionUtils.java]: Fixed.

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/InnerSpaceCompactionUtils.java [hunk 3]
```diff
@@ -108,12 +112,22 @@ private static void compactNotAlignedSeries(
     while (seriesIterator.hasNextSeries()) {
       checkThreadInterrupted(targetResource);
       // TODO: we can provide a configuration item to enable concurrent between each series
-      String currentSeries = seriesIterator.nextSeries();
+      PartialPath p = new PartialPath(device, seriesIterator.nextSeries());
```
[the remark this answers is not in the document]
Comment (ericpai) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/inner/utils/InnerSpaceCompactionUtils.java:115]: The logic of constructing the full timeseries path is not changed. 
Besides, the data of `seriesIterator` comes from https://github.com/apache/iotdb/blob/master/tsfile/src/main/java/org/apache/iotdb/tsfile/read/TsFileSequenceReader.java#L1896-L1931 , which returns a map with keys of type `TimeseriesMetaData.getMeasurementId()`.

**Review by qiaojialin — APPROVED** [review]
LGTM

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java [hunk 4]
```diff
@@ -121,20 +123,30 @@ private static void compactAlignedSeries(
...
-    List<String> allMeasurments = alignedMeasurmentIterator.getAllMeasurements();
+    List<String> allMeasurements = alignedMeasurmentIterator.getAllMeasurements();
     List<IMeasurementSchema> measurementSchemas = new ArrayList<>();
-    for (String measurement : allMeasurments) {
+    for (String measurement : allMeasurements) {
       // TODO: use IDTable
-      measurementSchemas.add(
-          IoTDB.metaManager.getSeriesSchema(new PartialPath(device, measurement)));
+      try {
+        measurementSchemas.add(
```
Comment (THUMarkLau) [inline server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java:131]: What if the series get deleted when compacting aligned series in inner space?

**Review by THUMarkLau — COMMENTED** [review]
Consider a aligned measurement is deleted when executing inner space compaction.

**Review by choubenson — APPROVED** [review]

File: server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java (hunk 4 above)
Comment (ericpai) [reply server/src/main/java/org/apache/iotdb/db/engine/compaction/CompactionUtils.java:131]: IMO, we can category the timing of deleting timeseries and compaction:
1. If the timeseries deletion happens before compaction, all the closed tsfiles will skip those deleted timeseries data and the merged one will not contain them.
2. If the timeseries deletion happens during compaction, as it has already checked the measurements validation before compaction, the deleted timeseries data is still merged in this compaction work. Although it's a waste of the resource, but no program or data error happens. And in next compaction task, the former deleted data will be processed as condition 1.

**Pull request merged**